src/components/ChannelList/hooks/listeners/useNewMessageNotification.js

```diff
@@ -0,0 +1,27 @@
+import { useContext, useEffect } from 'react';
+import uniqBy from 'lodash/uniqBy';
+
+import { ChatContext } from '../../../../context';
+import { getChannel } from '../../utils';
+
+export const useNewMessageNotification = ({ onMessageNew, setChannels }) => {
+  const { client } = useContext(ChatContext);
+
+  useEffect(() => {
+    const handleEvent = async (e) => {
+      if (onMessageNew && typeof onMessageNew === 'function') {
```

Comment:
do you actually need the first part as if it's null or undefined or 'falsy' it wouldn't pass the typeof check
```suggestion
      if (typeof onMessageNew === 'function') {
```